numpy/core/src/multiarray/ctors.c

```diff
@@ -2251,6 +2246,11 @@ PyArray_FromInterface(PyObject *origin)
             dtype = new_dtype;
         }
     }
+#if defined(NPY_PY3K)
+    if (tmp == attr) {
+        Py_DECREF(tmp);
+    }
+#endif
```

Comment:
This would be tidier as adding an extra `Py_INCREF` where we convert unicode to str in the py 2 case

Reply:
True, was still looking into other things warnings ;). Have to rush now, hopefully won't forget to check it some time tomorrow or so.

Reply:
Changed it, not sure it is much nicer, but also not worse (will push in a bit)